TestKubeletDefault unit test fails when run locally on system that has systemd-resolved active

**What happened**:
TestKubeletDefault unit test fails when run locally on system that has systemd-resolved active

**What you expected to happen**:
Unit test succeeds

**How to reproduce it (as minimally and precisely as possible)**:
Use ubuntu with systemd-resolved active.
Run `go test ./cmd/kubeadm/app/componentconfigs`
Output:
```

--- FAIL: TestKubeletDefault (0.01s)
    --- FAIL: TestKubeletDefault/No_specific_defaulting_works (0.00s)
        kubelet_test.go:328: Missmatch between expected and got:
            Expected:
            {{{ } /etc/kubernetes/manifests {0} {0} {0}  map[]  0 0   []  true false {{ca.crt} {0xc000041691 {0}} {0xc000041690}} {Webhook {{0} {0}}} <nil> 0 <nil> 0 <nil> false 0xc000041694 127.0.0.1 <nil>  [10.96.0.10] {0} {0} {0} 0 {0} <nil> <nil> {0}    <nil>   {0}  map[] {0}  0  <nil>  false <nil> <nil> 0  <nil> 0 <nil> map[] map[] map[] {0} 0 map[] 0 <nil> false <nil> <nil> <nil> map[] <nil>  <nil>  map[] map[]     [] [] }}
            ---
            Got:
            &{{{ } /etc/kubernetes/manifests {0} {0} {0}  map[]  0 0   []  true false {{ca.crt} {0xc000041a39 {0}} {0xc000041a38}} {Webhook {{0} {0}}} <nil> 0 <nil> 0 <nil> false 0xc000041a3c 127.0.0.1 <nil>  [10.96.0.10] {0} {0} {0} 0 {0} <nil> <nil> {0}    <nil>   {0}  map[] {0}  0  <nil> /run/systemd/resolve/resolv.conf false <nil> <nil> 0  <nil> 0 <nil> map[] map[] map[] {0} 0 map[] 0 <nil> false <nil> <nil> <nil> map[] <nil>  <nil>  map[] map[]     [] [] }}
    --- FAIL: TestKubeletDefault/Service_subnet,_no_dual_stack_defaulting_works (0.00s)
        kubelet_test.go:328: Missmatch between expected and got:
            Expected:
            {{{ } /etc/kubernetes/manifests {0} {0} {0}  map[]  0 0   []  true false {{ca.crt} {0xc000041699 {0}} {0xc000041698}} {Webhook {{0} {0}}} <nil> 0 <nil> 0 <nil> false 0xc00004169c 127.0.0.1 <nil>  [192.168.0.10] {0} {0} {0} 0 {0} <nil> <nil> {0}    <nil>   {0}  map[] {0}  0  <nil>  false <nil> <nil> 0  <nil> 0 <nil> map[] map[] map[] {0} 0 map[] 0 <nil> false <nil> <nil> <nil> map[] <nil>  <nil>  map[] map[]     [] [] }}
            ---
            Got:
            &{{{ } /etc/kubernetes/manifests {0} {0} {0}  map[]  0 0   []  true false {{ca.crt} {0xc0004fa1fd {0}} {0xc0004fa1fc}} {Webhook {{0} {0}}} <nil> 0 <nil> 0 <nil> false 0xc0004fa200 127.0.0.1 <nil>  [192.168.0.10] {0} {0} {0} 0 {0} <nil> <nil> {0}    <nil>   {0}  map[] {0}  0  <nil> /run/systemd/resolve/resolv.conf false <nil> <nil> 0  <nil> 0 <nil> map[] map[] map[] {0} 0 map[] 0 <nil> false <nil> <nil> <nil> map[] <nil>  <nil>  map[] map[]     [] [] }}
    --- FAIL: TestKubeletDefault/Service_subnet,_dual_stack_defaulting_works (0.00s)
        kubelet_test.go:328: Missmatch between expected and got:
            Expected:
            {{{ } /etc/kubernetes/manifests {0} {0} {0}  map[]  0 0   []  true false {{ca.crt} {0xc0000416a1 {0}} {0xc0000416a0}} {Webhook {{0} {0}}} <nil> 0 <nil> 0 <nil> false 0xc0000416a4 127.0.0.1 <nil>  [192.168.0.10] {0} {0} {0} 0 {0} <nil> <nil> {0}    <nil>   {0}  map[] {0}  0  <nil>  false <nil> <nil> 0  <nil> 0 <nil> map[] map[] map[] {0} 0 map[] 0 <nil> false <nil> <nil> <nil> map[] <nil>  <nil>  map[] map[]     [] [] }}
            ---
            Got:
            &{{{ } /etc/kubernetes/manifests {0} {0} {0}  map[]  0 0   []  true false {{ca.crt} {0xc0004fa3bd {0}} {0xc0004fa3bc}} {Webhook {{0} {0}}} <nil> 0 <nil> 0 <nil> false 0xc0004fa3c0 127.0.0.1 <nil>  [192.168.0.10] {0} {0} {0} 0 {0} <nil> <nil> {0}    <nil>   {0}  map[] {0}  0  <nil> /run/systemd/resolve/resolv.conf false <nil> <nil> 0  <nil> 0 <nil> map[] map[] map[] {0} 0 map[] 0 <nil> false <nil> <nil> <nil> map[] <nil>  <nil>  map[] map[]     [] [] }}
    --- FAIL: TestKubeletDefault/DNS_domain_defaulting_works (0.00s)
        kubelet_test.go:328: Missmatch between expected and got:
            Expected:
            {{{ } /etc/kubernetes/manifests {0} {0} {0}  map[]  0 0   []  true false {{ca.crt} {0xc0000416a9 {0}} {0xc0000416a8}} {Webhook {{0} {0}}} <nil> 0 <nil> 0 <nil> false 0xc0000416ac 127.0.0.1 <nil> example.com [10.96.0.10] {0} {0} {0} 0 {0} <nil> <nil> {0}    <nil>   {0}  map[] {0}  0  <nil>  false <nil> <nil> 0  <nil> 0 <nil> map[] map[] map[] {0} 0 map[] 0 <nil> false <nil> <nil> <nil> map[] <nil>  <nil>  map[] map[]     [] [] }}
            ---
            Got:
            &{{{ } /etc/kubernetes/manifests {0} {0} {0}  map[]  0 0   []  true false {{ca.crt} {0xc0004fa539 {0}} {0xc0004fa538}} {Webhook {{0} {0}}} <nil> 0 <nil> 0 <nil> false 0xc0004fa53c 127.0.0.1 <nil> example.com [10.96.0.10] {0} {0} {0} 0 {0} <nil> <nil> {0}    <nil>   {0}  map[] {0}  0  <nil> /run/systemd/resolve/resolv.conf false <nil> <nil> 0  <nil> 0 <nil> map[] map[] map[] {0} 0 map[] 0 <nil> false <nil> <nil> <nil> map[] <nil>  <nil>  map[] map[]     [] [] }}
    --- FAIL: TestKubeletDefault/CertificatesDir_defaulting_works (0.00s)
        kubelet_test.go:328: Missmatch between expected and got:
            Expected:
            {{{ } /etc/kubernetes/manifests {0} {0} {0}  map[]  0 0   []  true false {{/path/to/certs/ca.crt} {0xc0000418c1 {0}} {0xc0000418c0}} {Webhook {{0} {0}}} <nil> 0 <nil> 0 <nil> false 0xc0000418c4 127.0.0.1 <nil>  [10.96.0.10] {0} {0} {0} 0 {0} <nil> <nil> {0}    <nil>   {0}  map[] {0}  0  <nil>  false <nil> <nil> 0  <nil> 0 <nil> map[] map[] map[] {0} 0 map[] 0 <nil> false <nil> <nil> <nil> map[] <nil>  <nil>  map[] map[]     [] [] }}
            ---
            Got:
            &{{{ } /etc/kubernetes/manifests {0} {0} {0}  map[]  0 0   []  true false {{/path/to/certs/ca.crt} {0xc0004fa6b9 {0}} {0xc0004fa6b8}} {Webhook {{0} {0}}} <nil> 0 <nil> 0 <nil> false 0xc0004fa6bc 127.0.0.1 <nil>  [10.96.0.10] {0} {0} {0} 0 {0} <nil> <nil> {0}    <nil>   {0}  map[] {0}  0  <nil> /run/systemd/resolve/resolv.conf false <nil> <nil> 0  <nil> 0 <nil> map[] map[] map[] {0} 0 map[] 0 <nil> false <nil> <nil> <nil> map[] <nil>  <nil>  map[] map[]     [] [] }}
FAIL
FAIL	k8s.io/kubernetes/cmd/kubeadm/app/componentconfigs	0.032s
FAIL
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): **Master branch (v1.19.0-alpha)**
- Cloud provider or hardware configuration: **n/a**
- OS (e.g: `cat /etc/os-release`):
```
NAME="Ubuntu"
VERSION="20.04 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```
- Kernel (e.g. `uname -a`): 
```
Linux cinlogic-xps13 5.4.0-28-generic #32-Ubuntu SMP Wed Apr 22 17:40:10 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:
```
$ systemd --version
systemd 245 (245.4-4ubuntu3)
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid
```
```
$ systemctl status systemd-resolved
● systemd-resolved.service - Network Name Resolution
     Loaded: loaded (/lib/systemd/system/systemd-resolved.service; enabled; vendor preset: enabled)
     Active: active (running) since Mon 2020-05-04 10:12:56 EDT; 1h 32min ago
       Docs: man:systemd-resolved.service(8)
             https://www.freedesktop.org/wiki/Software/systemd/resolved
             https://www.freedesktop.org/wiki/Software/systemd/writing-network-configuration-managers
             https://www.freedesktop.org/wiki/Software/systemd/writing-resolver-clients
   Main PID: 881 (systemd-resolve)
     Status: "Processing requests..."
      Tasks: 1 (limit: 18780)
     Memory: 4.0M
     CGroup: /system.slice/systemd-resolved.service
             └─881 /lib/systemd/systemd-resolved
...
```
```
$ ll /run/systemd/resolve/resolv.conf
-rw-r--r-- 1 systemd-resolve systemd-resolve 615 May  4 10:13 /run/systemd/resolve/resolv.conf
```

--- a/cmd/kubeadm/app/componentconfigs/kubelet_test.go
+++ b/cmd/kubeadm/app/componentconfigs/kubelet_test.go
@@ -155,6 +155,12 @@ func TestKubeletUnmarshal(t *testing.T) {
 }
 
 func TestKubeletDefault(t *testing.T) {
+	var resolverConfig string
+	if isSystemdResolvedActive, _ := isServiceActive("systemd-resolved"); isSystemdResolvedActive {
+		// If systemd-resolved is active, we need to set the default resolver config
+		resolverConfig = kubeletSystemdResolverConfig
+	}
+
 	tests := []struct {
 		name       string
 		clusterCfg kubeadmapi.ClusterConfiguration
@@ -185,6 +191,7 @@ func TestKubeletDefault(t *testing.T) {
 					HealthzBindAddress: kubeletHealthzBindAddress,
 					HealthzPort:        utilpointer.Int32Ptr(constants.KubeletHealthzPort),
 					RotateCertificates: kubeletRotateCertificates,
+					ResolverConfig:     resolverConfig,
 				},
 			},
 		},
@@ -217,6 +224,7 @@ func TestKubeletDefault(t *testing.T) {
 					HealthzBindAddress: kubeletHealthzBindAddress,
 					HealthzPort:        utilpointer.Int32Ptr(constants.KubeletHealthzPort),
 					RotateCertificates: kubeletRotateCertificates,
+					ResolverConfig:     resolverConfig,
 				},
 			},
 		},
@@ -252,6 +260,7 @@ func TestKubeletDefault(t *testing.T) {
 					HealthzBindAddress: kubeletHealthzBindAddress,
 					HealthzPort:        utilpointer.Int32Ptr(constants.KubeletHealthzPort),
 					RotateCertificates: kubeletRotateCertificates,
+					ResolverConfig:     resolverConfig,
 				},
 			},
 		},
@@ -285,6 +294,7 @@ func TestKubeletDefault(t *testing.T) {
 					HealthzBindAddress: kubeletHealthzBindAddress,
 					HealthzPort:        utilpointer.Int32Ptr(constants.KubeletHealthzPort),
 					RotateCertificates: kubeletRotateCertificates,
+					ResolverConfig:     resolverConfig,
 				},
 			},
 		},
@@ -315,6 +325,7 @@ func TestKubeletDefault(t *testing.T) {
 					HealthzBindAddress: kubeletHealthzBindAddress,
 					HealthzPort:        utilpointer.Int32Ptr(constants.KubeletHealthzPort),
 					RotateCertificates: kubeletRotateCertificates,
+					ResolverConfig:     resolverConfig,
 				},
 			},
 		},
@@ -325,7 +336,7 @@ func TestKubeletDefault(t *testing.T) {
 			got := &kubeletConfig{}
 			got.Default(&test.clusterCfg, &kubeadmapi.APIEndpoint{}, &kubeadmapi.NodeRegistrationOptions{})
 			if !reflect.DeepEqual(got, &test.expected) {
-				t.Fatalf("Missmatch between expected and got:\nExpected:\n%v\n---\nGot:\n%v", test.expected, got)
+				t.Fatalf("Missmatch between expected and got:\nExpected:\n%v\n---\nGot:\n%v", test.expected, *got)
 			}
 		})
 	}

